packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -123,11 +133,19 @@ function Modal(props: ModalPropsType): ReactElement {
 										</span>
 
 										{secondaryAction && (
-											<ButtonSecondary {...secondaryAction} data-testid="modal.buttons.secondary" />
+											<ButtonSecondary
+												{...secondaryAction}
+												data-testid="modal.buttons.secondary"
+												data-feature="modal.buttons.secondary"
```

Comment:
Can't we pass the data-feature we want to with the spread of secondaryAction driven by the host ?